Presentations/02-intro-to-R-programming.Rmd

```diff
@@ -1112,224 +1088,284 @@ name: appendix
+We didn't cover it here because of time constraints, but below are some useful links, and DIME Analytics provides trainings on Git and GitHub, so keep an eye out for them.
 
-* Now create an if statement using this switch scalar as a switch to indicate if you want to install a set of packages:
+* **DIME Analytics git page:** https://worldbank.github.io/dimeanalytics/git/
 
-```{r, eval = F}
+* **A Quick Introduction to Version Control with Git and GitHub:** https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004668
 
-    # Turn switch on
-    PACKAGES <- 1
+---
 
-    # Install packages
-    if (PACKAGES == 1) {
-      install.packages(packages,
-                       dependencies = TRUE)
-    }
+# Appendix - More on R projects
 
-```
+## R projects
+
+If you want to learn more about them, we recommend starting here:
+https://r4ds.had.co.nz/workflow-projects.html
 
 ---
 
-# Appendix - If statements
+# Appendix - Commenting
 
-## If statements
+* To comment a line, write `#` as its first character
 
-Possible variations would include
+```{r, eval=FALSE}
+# This is a comment
+print("But this part is not")
+```
 
-```{r, eval = F}
-    # Turn switch on
-    PACKAGES <- TRUE
+* You can also add `#` halfway through a line to comment whatever comes after it
 
-    # Using a Boolean object
-    if (PACKAGES == TRUE) {
-      install.packages(packages, dep = TRUE)
-    }
+```{r, eval=FALSE}
+print("This part is not a comment") # And this is a comment
+```
 
-    # Which is the same as
-    if (PACKAGES) {
-      install.packages(packages, dep = TRUE)
-    }
+* In Stata, you can use `/*` and `*/` to comment in the middle of a line's code. That is not possible in R: everything that comes after `#` will always be a comment
 
-```
+* To comment a selection of lines, press `Ctrl` + `Shift` + `C`
 
 ---
 
-# Appendix - Assignment 2
+# Appendix - Assignment 3
 
-### Exercise: Create a function that...
+.exercise[
 
-1. Takes as argument a vector of packages names
+### Exercise  `r fa("keyboard")`
 
-2. Loops through the packages listed in the input vector
+1. In your script panel, select all the lines of your script
 
-3. Tests if a package is already installed
+2. Use the keyboard shortcut to comment these lines.
+  + Shortcut: `Ctrl` + `Shift` + `C`
 
-4. Only installs packages that are not yet installed
+3. Use the keyboard shortcut to comment these lines again. What happened?
 
-5. Loads the packages
+]
+
+---
+
+# Appendix - Looping
+
+## Loops
+
+* In Stata, we use `for` loops very frequently
+
+* The equivalent to that in R would be this
 
-- TIP: to test if a package is already installed, use the following code:
 ```{r, eval = F}
-    # Test if object x is contained in
-    # the vector of installed packages
-    x %in% installed.packages()
+    # A for loop in R
+    for (number in 1:5) {
+        print(number)
+    }
 ```
 
 ---
 
-# Appendix - Other file path practices
+# Appendix - Looping
```

Comment:
I think everything after this slide should go. These slides are old, are not following best practices and solve problems for which we have better solutions now.

Reply:
agreed. I see you already deleted these contents in 8e197aa